src/shared/dropin.c

```diff
@@ -128,14 +128,12 @@ static int unit_file_find_dir(
         assert(path);
 
         r = chase_symlinks(path, original_root, 0, &chased);
-        /* Ignore -ENOENT, after all most units won't have a drop-in dir.
-         * Also ignore -ENAMETOOLONG, users are not even able to create
-         * the drop-in dir in such case. This mostly happens for device units with long /sys path.
-         * */
-        if (IN_SET(r, -ENOENT, -ENAMETOOLONG))
+        if (r == -ENOENT) /* Ignore -ENOENT, after all most units won't have a drop-in dir. */
                 return 0;
-        if (r < 0)
-                return log_full_errno(LOG_WARNING, r, "Failed to canonicalize path %s: %m", path);
+        if (r < 0) /* Also, downgrade logging about -ENAMETOOLONG to LOG_DEBUG, users are not even able to create the
+                    * drop-in dir in such case. This mostly happens for device units with long /sys path. */
+                return log_full_errno(r == -ENAMETOOLONG ? LOG_DEBUG : LOG_WARNING, r,
+                                      "Failed to canonicalize path %s: %m", path);
```

Comment:
That changes the behaviour, because an error is returned, which causes unit_file_find_dirs() to abort early.

Reply:
indeed